src/test/groovy/GithubCheckStepTests.groovy

```diff
@@ -0,0 +1,234 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import org.junit.Before
+import org.junit.Test
+import static org.junit.Assert.assertFalse
+import static org.junit.Assert.assertNull
+import static org.junit.Assert.assertSame
+import static org.junit.Assert.assertTrue
+
+class GithubCheckStepTests extends ApmBasePipelineTest {
+  def script
+
+  def privateKey
+
+  @Override
+  @Before
+  void setUp() throws Exception {
+    super.setUp()
+    script = loadScript('vars/githubCheck.groovy')
+    privateKey = new File("src/test/resources/github-app-private-key-tests.pem").text
```

Comment:
I haven't checked this in the past, but maybe this initialisation could be done at BeforeClass level, to execute this block only once. This could be refactored in a follow-up PR.

Reply:
That's a good point, I have not tried though, let me do it

Reply:
I've tried the BeforeClass locally but it was not as intuitive as I though, I'll probably work on a follow-up